src/pages/commits/HistoryTableIcon/index.tsx

```diff
@@ -0,0 +1,65 @@
+import styled from "@emotion/styled";
+import Tooltip from "@leafygreen-ui/tooltip";
+import { Body } from "@leafygreen-ui/typography";
+import { TaskStatusIcon } from "components/TaskStatusIcon";
+import { TaskStatus } from "types/task";
+
+interface HistoryTableIconProps {
+  status: TaskStatus;
+  label?: string;
+  failingTests?: string[];
+  inactive?: boolean;
+  onClick?: () => void;
+}
+
+export const HistoryTableIcon: React.FC<HistoryTableIconProps> = ({
+  status,
+  label,
+  failingTests = [],
+  inactive,
+  onClick,
+}) => (
+  <Tooltip
+    usePortal={false}
+    align="right"
+    justify="middle"
+    enabled={!inactive && !!failingTests.length}
+    popoverZIndex={1}
+    trigger={
+      <Container onClick={onClick}>
+        <IconContainer inactive={inactive}>
+          <TaskStatusIcon status={status} />
+        </IconContainer>
+        {!inactive && <Body>{label}</Body>}
+      </Container>
+    }
+    triggerEvent="hover"
+  >
+    <TestNameContainer>
+      {failingTests.map((testName) => (
+        <Body>{testName}</Body>
+      ))}
+    </TestNameContainer>
+  </Tooltip>
+);
+
+const TestNameContainer = styled.div`
+  white-space: nowrap;
+`;
+
+interface ContainerProps {
+  onClick?: () => void;
+}
+const Container = styled.div<ContainerProps>`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  ${({ onClick }) => onClick && "cursor: pointer;"}
+`;
+
+interface IconContainerProps {
+  inactive?: boolean;
+}
+const IconContainer = styled.div<IconContainerProps>`
+  ${({ inactive }) => inactive && "opacity: .4;"}
```

Comment:
[nit] It seems like this same inactive styling appears [elsewhere](https://github.com/evergreen-ci/spruce/blob/cf94d556699ba8d6f862c3cc5be3b50dd7b5c619/src/pages/commits/ActiveCommits/CommitChartTooltip.tsx#L98), could be nice to make it a reusable style